package/MDAnalysis/analysis/density.py

```diff
@@ -383,7 +384,18 @@ def __init__(self, atomgroup, delta=1.0,
         self._ydim = ydim
         self._zdim = zdim
 
+    def _checkinput(self):
+        if len(self._atomgroup) == 0:
+            msg = ("No atoms in selection over whole trajectory")
```

Comment:
Is this a correct statement? If you're passed an `UpdatingAtomGroup`, does checking only the first frame tell you that there will be no atoms covering this particular selection for the entirety of the trajectory?

Reply:
Yeah, perhaps this should be - "No atoms in selection in frame #number"